--- firmware-addon-dell-1.4.10/Makefile.old	2007-12-09 17:42:39.000000000 -0600
+++ firmware-addon-dell-1.4.10/Makefile	2007-12-09 17:42:45.000000000 -0600
@@ -158,6 +158,10 @@ ifeq ($(NEEDS_DIST), 1)
   ifndef DIST
   $(error "Must set DIST={gutsy,hardy,sid,...} for deb and sdeb targets")
   endif
+  ifndef DISTTAG
+  $(info Remember to set DISTTAG='~gutsy1' for deb and sdeb targets for backports)
+  DISTTAG =
+  endif
 endif
 
 deb: $(TARBALL)
@@ -165,25 +169,29 @@ deb: $(TARBALL)
 	tmp_dir=`mktemp -d /tmp/firmware-addon-dell.XXXXXXXX` ; \
 	tar zxvf $(TARBALL) -C $${tmp_dir} ; \
 	cp $(TARBALL) $${tmp_dir}/$(RELEASE_NAME)_$(RELEASE_VERSION).orig.tar.gz ; \
-	cp -a pkg/debian $${tmp_dir}/$(RELEASE_STRING)/ ; \
+	cp -ar $(BUILDDIR)/pkg/debian $${tmp_dir}/$(RELEASE_STRING)/debian ; \
 	chmod +x $${tmp_dir}/$(RELEASE_STRING)/debian/rules ; \
-	sed -e "s/#DIST#/$(DIST)/g" $${tmp_dir}/$(RELEASE_STRING)/debian/changelog.in > $${tmp_dir}/$(RELEASE_STRING)/debian/changelog ; \
+	sed -e "s/#DISTTAG#/$(DISTTAG)/g" -e "s/#DIST#/$(DIST)/g" $${tmp_dir}/$(RELEASE_STRING)/debian/changelog.in > $${tmp_dir}/$(RELEASE_STRING)/debian/changelog ; \
+	rm $${tmp_dir}/$(RELEASE_STRING)/debian/changelog.in ; \
 	cd $${tmp_dir}/$(RELEASE_STRING)/ ; \
-	pdebuild --use-pdebuild-internal --auto-debsign --buildresult $(deb_destdir) ; \
+	mkdir -p $(deb_destdir)/$(DIST) ; \
+	pdebuild --use-pdebuild-internal --buildresult $(deb_destdir)/$(DIST) ; \
 	cd - ; \
 	rm -rf $${tmp_dir}
 
-sdeb: $(TARBALL) dist_set
+sdeb: $(TARBALL)
 	mkdir -p $(deb_destdir) ; \
 	tmp_dir=`mktemp -d /tmp/firmware-addon-dell.XXXXXXXX` ; \
 	cp $(TARBALL) $${tmp_dir}/$(RELEASE_NAME)_$(RELEASE_VERSION).orig.tar.gz ;\
 	tar -C $${tmp_dir} -xzf $(TARBALL) ; \
-	mv $${tmp_dir}/$(RELEASE_STRING)/pkg/debian $${tmp_dir}/$(RELEASE_STRING)/debian ; \
+	cp -ar $(BUILDDIR)/pkg/debian $${tmp_dir}/$(RELEASE_STRING)/debian ; \
 	chmod +x $${tmp_dir}/$(RELEASE_STRING)/debian/rules ; \
-	sed -e "s/#DIST#/$(DIST)/g" $${tmp_dir}/$(RELEASE_STRING)/debian/changelog.in > $${tmp_dir}/$(RELEASE_STRING)/debian/changelog ; \
+	sed -e "s/#DISTTAG#/$(DISTTAG)/g" -e "s/#DIST#/$(DIST)/g" $${tmp_dir}/$(RELEASE_STRING)/debian/changelog.in > $${tmp_dir}/$(RELEASE_STRING)/debian/changelog ; \
+	rm $${tmp_dir}/$(RELEASE_STRING)/debian/changelog.in ; \
 	cd $${tmp_dir}/$(RELEASE_STRING) ; \
 	dpkg-buildpackage -D -S -sa -rfakeroot ; \
-	mv ../$(RELEASE_NAME)_* $(deb_destdir) ; \
+	mkdir -p $(deb_destdir)/$(DIST) ; \
+	mv ../$(RELEASE_NAME)_* $(deb_destdir)/$(DIST) ; \
 	cd - ;\
 	rm -rf $${tmp_dir}
 
